### DISCUSSION TITLE
http   deviceprofile get

### DISCUSSION THREAD

**Discussion by wangqingpen** [edgexfoundry/community#229] [Q&A]
Can I change the value of the attributes parameter in the deviceprofile when I call the get command through the http protocol?

**Comment by jiekechoo** [comment]
@wangqingpen , here is the API-Reference: https://docs.edgexfoundry.org/3.1/microservices/core/metadata/ApiReference/

 Try this API `/deviceprofile` to update existing device profile.
<img width="1451" alt="image" src="https://github.com/edgexfoundry/community/assets/3350211/390745fc-ab83-477e-9439-7bebcf9b01c7">

**Reply by wangqingpen** [reply]
https://docs.edgexfoundry.org/3.1/microservices/core/metadata/ApiReference/           
The webpage opens  is           
![image](https://github.com/edgexfoundry/community/assets/72652818/43dc6e77-6d5d-475b-9ef0-886c8dc87f39)

**Reply by jiekechoo** [reply]
Maybe you need a network to access the domain `raw.githubusercontent.com`.

**Reply by wangqingpen** [reply]
My StatusCode: 400 json is not meant to contain all the DeviceResources commands

![image](https://github.com/edgexfoundry/community/assets/72652818/566d0419-ca43-4fe3-9977-08d5fdc7aba1)

**Reply by wangqingpen** [reply]
The above problem has been solved, and there is a new problem at present, and the return value is 200, but my proflie file has not changed, is my uRL written incorrectly? The profile file is located as follows 
![image](https://github.com/edgexfoundry/community/assets/72652818/63afd3ae-27d4-42ee-b48b-b2c8b269b906)

 URL is  "http://192.168.12.77:4000/api/v2/deviceprofile"

**Reply by jiekechoo** [reply]
Are you using EdgeX v2? Please upgrade to v3, v3.1.0(Napa) is perferred. It just released a month ago.

**Reply by wangqingpen** [reply]
Is it not supported by v2

**Reply by wangqingpen** [reply]
I changed the URL to url := "http://192.168.12.77:59881/api/v2/deviceprofile" 
I got 
Response: {"apiVersion":"v2","message":"Failed to unmarshal request body as JSON.","statusCode":400}                 
 Is it not supported by the v2 version?

**Comment by cloudxxx8** [comment]
Could you please specify what you want to change?
The Get Command execution is only for calling Get Command, not updating any profile content.
The question is not clear to me.
